sse.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2022 Flamego. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package sse
+
+import (
+	"encoding/json"
+	"net/http"
+	"reflect"
+	"time"
+
+	"github.com/flamego/flamego"
+)
+
+const (
+	defaultPingInterval = 10 * time.Second
+)
+
+type Options struct {
+	// PingInterval is the time to wait between sending pings to the client.
+	PingInterval time.Duration
+}
+
+type Connection struct {
+	*Options
+
+	// Sender is the channel used for sending out data to the client.
+	// This channel gets mapped for the next handler to use with the right type
+	// and is asynchronous unless the SendChannelBuffer is set to 0.
+	Sender reflect.Value
+
+	// context is the request context.
+	context flamego.Context
+
+	// ticker for pinging the client.
+	ticker *time.Ticker
+}
+
+func Handler(bind interface{}, options ...*Options) flamego.Handler {
```

Comment:
Why do we need pointer to the `Options`? Doesn't seem like being consistent with other middleware, unless we have a very good reason to do so.